tests/unit/test_config.py

```diff
@@ -0,0 +1,170 @@
+    )
+
+
+def test_invalid_config() -> None:
+    """Check that invalid configuration is handled gracefully."""
+    check_expected_exception("""""", InvalidConfigurationError, "no LLMConfig found")
+    check_expected_exception(
+        """{foo=123}""", InvalidConfigurationError, "no LLMConfig found"
+    )
+    check_expected_exception(
+        """
+---
+LLMProviders:
+  - name: p1
+    url: url1
+    credentials: key1
+    models:
+      - name: m1
+        url: murl1
+        credentials: mkey1
+      - name: m2
+        url: murl2
+        credentials: mkey2
+  - name: p2
+    url: url2
+    credentials: key2
+    models:
+      - name: m1
+        url: murl1
+        credentials: mkey1
+      - name: m2
+        url: murl2
+        credentials: mkey2
+""",
+        InvalidConfigurationError,
+        "no OLSConfig found",
+    )
+
+    check_expected_exception(
+        """
+---
+LLMProviders:
+  - foo: p1
+    url: url1
+    credentials: key1
+    models:
+      - name: m1
+        url: murl1
+        credentials: mkey1
+""",
+        InvalidConfigurationError,
+        "provider name is missing",
+    )
+
+
+def test_valid_config_stream() -> None:
+    """Check if a valid configuration stream is handled correctly."""
+    try:
+        LLMConfigYaml = """
+---
+LLMProviders:
+  - name: p1
+    url: url1
+    credentials: key1
+    models:
+      - name: m1
+        url: murl1
+        credentials: mkey1
+      - name: m2
+        url: murl2
+        credentials: mkey2
+  - name: p2
+    url: url2
+    credentials: key2
+    models:
+      - name: m1
+        url: murl1
+        credentials: mkey1
+      - name: m2
+        url: murl2
+        credentials: mkey2
+OLSConfig:
+  enable_debug_ui: false
+  conversation_cache:
+    type: in-memory
+    redis:
+      url: 127.0.0.1:1234
+      credentials:
+        user: root
+        password: pwd123
+    in-memory:
+      max_entries: 1000
+  logger_config:
+    level: info
+  default_provider: p1
+  default_model: m1
+  classifier_provider: p2
+  classifier_model: m1
+"""
+        config.load_config_from_stream(io.StringIO(LLMConfigYaml))
+    except Exception:
+        print(traceback.format_exc())
+        assert False
+
+
+def test_valid_config_file() -> None:
+    """Check if a valid configuration file is handled correctly."""
+    try:
+        config.load_config("olsconfig.yaml")
+        # it would be great to roundtrip the config back to YAML and diff the two YAMLs
+        assert len(config.config.llm_config.providers) == 2
+        p1 = config.config.llm_config.providers["p1"]
+        assert p1.name == "p1"
+        assert p1.url == "url1"
+        assert p1.credential_path == "secret.txt"
+        assert p1.credentials == "secret_key"
+        assert len(p1.models) == 2
+        m1 = p1.models["m1"]
+        assert m1.name == "m1"
+        assert m1.url == "murl1"
+        assert m1.credentials == "mkey1"
```

Comment:
if we're going to have a test like this it should test all the fields of the config (so the ols config fields too, not just the llm config ones)